### PR TITLE
test(internal/librarian/rust): add missing test coverage for update_manifest.go

### PR DESCRIPTION
**Pull request by jameslynnwu**
Unit tests are added for updateCargoVersion and updateWorkspaceVersion in update_manifest_test.go to achieve 100 percent coverage for update_manifest.go.

These functions were previously untested. The new tests cover success cases, file not found errors, and specific edge cases like missing version fields in Cargo.toml.

### CODE REVIEW

File: internal/librarian/rust/update_manifest_test.go [hunk 1]
```diff
@@ -96,3 +102,123 @@ func TestShouldBumpManifestVersionBadDiff(t *testing.T) {
...
+	}
+
+	updatedContent, err := os.ReadFile(filePath)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	expected := "[package]\nname = \"test-crate\"\nversion                = \"2.0.0\"\n"
+	if diff := cmp.Diff(expected, string(updatedContent)); diff != "" {
+		t.Errorf("mismatch got = %q, want %q", string(updatedContent), expected)
```
Comment (JoeWang1127) [inline internal/librarian/rust/update_manifest_test.go]: Use standard error format: https://github.com/googleapis/librarian/blob/7bf0e08615d35f0c412274b6f420cca5951eea46/doc/howwewritego.md#use-cmpdiff-for-comparisons

File: internal/librarian/rust/update_manifest_test.go [hunk 2]
```diff
@@ -96,3 +102,123 @@ func TestShouldBumpManifestVersionBadDiff(t *testing.T) {
...
+		},
+		{
+			name:    "file not found",
+			content: "",
+			wantErr: fs.ErrNotExist,
+		},
+	} {
+		t.Run(test.name, func(t *testing.T) {
+			var filePath string
+			if test.content != "" {
```
Comment (JoeWang1127) [inline internal/librarian/rust/update_manifest_test.go]: nit: use a setup function to make the test case straightforward.

File: internal/librarian/rust/update_manifest_test.go [hunk 3]
```diff
@@ -96,3 +102,123 @@ func TestShouldBumpManifestVersionBadDiff(t *testing.T) {
...
+				filePath = path.Join(dir, "Cargo.toml")
+				if err := os.WriteFile(filePath, []byte(test.content), 0644); err != nil {
+					t.Fatal(err)
+				}
+			} else {
+				filePath = "non-existent-file"
+			}
+
+			newVersion, _ := semver.Parse("2.0.0")
+
```
Comment (JoeWang1127) [inline internal/librarian/rust/update_manifest_test.go]: You don't need this empty line.

File: internal/librarian/rust/update_manifest_test.go [hunk 4]
```diff
@@ -96,3 +102,123 @@ func TestShouldBumpManifestVersionBadDiff(t *testing.T) {
...
+			if test.content != "" {
+				dir := t.TempDir()
+				filePath = path.Join(dir, "Cargo.toml")
+				if err := os.WriteFile(filePath, []byte(test.content), 0644); err != nil {
+					t.Fatal(err)
+				}
+			} else {
+				filePath = "non-existent-file"
+			}
+
```
Comment (JoeWang1127) [inline internal/librarian/rust/update_manifest_test.go]: You don't need this empty line.

File: internal/librarian/rust/update_manifest_test.go [hunk 5]
```diff
@@ -96,3 +102,123 @@ func TestShouldBumpManifestVersionBadDiff(t *testing.T) {
...
+				t.Fatal(err)
+			}
+
+			updatedContent, err := os.ReadFile(filePath)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			if diff := cmp.Diff(test.want, string(updatedContent)); diff != "" {
+				t.Errorf("mismatch got = %q, want %q", string(updatedContent), test.want)
```
Comment (JoeWang1127) [inline internal/librarian/rust/update_manifest_test.go]: Use standard error format: https://github.com/googleapis/librarian/blob/7bf0e08615d35f0c412274b6f420cca5951eea46/doc/howwewritego.md#use-cmpdiff-for-comparisons

File: internal/librarian/rust/update_manifest_test.go [hunk 6]
```diff
@@ -96,3 +102,123 @@ func TestShouldBumpManifestVersionBadDiff(t *testing.T) {
...
+			if err := os.WriteFile(filePath, []byte(test.content), 0644); err != nil {
+				t.Fatal(err)
+			}
+
+			newVersion, _ := semver.Parse("2.0.0")
+
+			if err := updateWorkspaceVersion(filePath, test.crateName, newVersion); err != nil {
+				t.Fatal(err)
+			}
+
```
Comment (JoeWang1127) [inline internal/librarian/rust/update_manifest_test.go]: You don't need this empty line.

File: internal/librarian/rust/update_manifest_test.go [hunk 7]
```diff
@@ -96,3 +102,123 @@ func TestShouldBumpManifestVersionBadDiff(t *testing.T) {
...
+			want:      "[workspace.dependencies]\nother-crate = { version = \"1.0.0\", path = \"src/other-crate\" }\n",
+		},
+	} {
+		t.Run(test.name, func(t *testing.T) {
+			dir := t.TempDir()
+			filePath := path.Join(dir, "Cargo.toml")
+			if err := os.WriteFile(filePath, []byte(test.content), 0644); err != nil {
+				t.Fatal(err)
+			}
+
```
Comment (JoeWang1127) [inline internal/librarian/rust/update_manifest_test.go]: You don't need this empty line.

**Review by JoeWang1127 — APPROVED** [review]
